viewing_party/party.py

```diff
@@ -1,23 +1,172 @@
+        for movie in range(len(list(friends_watched[friends]["watched"]))):
+            all_movies.append(friends_watched[friends]["watched"][movie]["title"])
+        
+
+    for movie in range(len(watched)):
+        if watched[movie]["title"] not in all_movies:
+
+            unique.append(watched[movie])
+            
+    return unique
+
+def get_friends_unique_watched(user_data):
+    
+    watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    all_movies = []
+    for movies in range(len(list(watched))):
+        all_movies.append(watched[movies]["title"])
+    friends_movies = []
+    for movie in range(len(friends_watched)):
+    
+        friends_movies.append(friends_watched[movie]["watched"])
+    print(f"all_movies_friends are equal to {friends_movies}")
+    result = []
+    for movies in range(len(friends_movies)):
+        for name in range(len(friends_movies[movie])):
+            if friends_movies[movies][name]["title"] not in all_movies:
+                result.append(friends_movies[movies][name])
+          
+
+    return result
+
+        
 
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    
+    watched_user = []
+    for movie in range(len(list(watched))):
+        watched_user.append(watched[movie]["title"])
+    
+
+    results = []
+    for i in range(len(list(friends_watched[0]["watched"]))):
+        if friends_watched[0]["watched"][i]["title"] not in watched_user and friends_watched[0]["watched"][i]["host"] in user_data["subscriptions"]:
+            results.append(friends_watched[0]["watched"][i])
+        
+    print(results)
+    return results
+    
+    
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+from collections import Counter
+def   get_new_rec_by_genre(user_data):
+    if not user_data["watched"] and user_data["friends"]:
+        return []
+    watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    genre = []
+    watched_user = []
+    for movie in range(len(list(watched))):
+        watched_user.append(watched[movie]["title"])
+        genre.append(watched[movie]["genre"])
+    favorite = Counter(genre)
+    fave_max = max(favorite, key=favorite.get)
+    results = []
+
+    for i in range(len(list(friends_watched[0]["watched"]))):
+        if friends_watched[0]["watched"][i]["title"] not in watched_user and friends_watched[0]["watched"][i]["genre"] == fave_max:
+                results.append(friends_watched[0]["watched"][i])
+   
+
+    return results
+
+def get_rec_from_favorites(user_data):
+    if not user_data["favorites"] and user_data["friends"]:
+        return []
+    favorites = user_data["favorites"]
+    
+    all = []

```

Comment:
`all` can also be mistaken as the `all()` function. Perhaps `all_movies_list` or `all_movies` would be a good name for this scenario?